app/uk/gov/hmrc/individualsifapistub/domain/Income.scala

```diff
@@ -16,6 +16,409 @@
+        (JsPath \ "inPayPeriod4").readNullable[Double](paymentAmountValidator)
+      ) (GrossEarningsForNics.apply _),
+    (
+      (JsPath \ "inPayPeriod1").writeNullable[Double] and
+        (JsPath \ "inPayPeriod2").writeNullable[Double] and
+        (JsPath \ "inPayPeriod3").writeNullable[Double] and
+        (JsPath \ "inPayPeriod4").writeNullable[Double]
+      ) (unlift(GrossEarningsForNics.unapply))
+  )
+
+  implicit val employeeNicsFormat: Format[EmployeeNics] = Format(
+    (
+      (JsPath \ "inPayPeriod1").readNullable[Double](paymentAmountValidator) and
+        (JsPath \ "inPayPeriod2").readNullable[Double](paymentAmountValidator) and
+        (JsPath \ "inPayPeriod3").readNullable[Double](paymentAmountValidator) and
+        (JsPath \ "inPayPeriod4").readNullable[Double](paymentAmountValidator) and
+        (JsPath \ "ytd1").readNullable[Double](paymentAmountValidator) and
+        (JsPath \ "ytd2").readNullable[Double](paymentAmountValidator) and
+        (JsPath \ "ytd3").readNullable[Double](paymentAmountValidator) and
+        (JsPath \ "ytd4").readNullable[Double](paymentAmountValidator)
+      ) (EmployeeNics.apply _),
+    (
+      (JsPath \ "inPayPeriod1").writeNullable[Double] and
+        (JsPath \ "inPayPeriod2").writeNullable[Double] and
+        (JsPath \ "inPayPeriod3").writeNullable[Double] and
+        (JsPath \ "inPayPeriod4").writeNullable[Double] and
+        (JsPath \ "ytd1").writeNullable[Double] and
+        (JsPath \ "ytd2").writeNullable[Double] and
+        (JsPath \ "ytd3").writeNullable[Double] and
+        (JsPath \ "ytd4").writeNullable[Double]
+      ) (unlift(EmployeeNics.unapply))
+  )
+
+  implicit val payeEntryFormat: Format[PayeEntry] = Format(
+    (
+      (JsPath \ "taxCode").readNullable[String](minLength[String](2).keepAnd(maxLength[String](7).keepAnd(pattern(taxCodePattern, "Invalid Tax Code")))) and
+        (JsPath \ "paidHoursWorked").readNullable[String](maxLength[String](35).keepAnd(pattern(paidHoursWorkPattern, "Invalid Paid Hours Work"))) and
+        (JsPath \ "taxablePayToDate").readNullable[Double](paymentAmountValidator) and
+        (JsPath \ "totalTaxToDate").readNullable[Double](paymentAmountValidator) and
+        (JsPath \ "taxDeductedOrRefunded").readNullable[Double](paymentAmountValidator) and
+        (JsPath \ "employerPayeRef").readNullable[String](maxLength[String](10).keepAnd(pattern(employerPayeRefPattern, "Invalid employer PAYE reference"))) and
+        (JsPath \ "paymentDate").readNullable[String](pattern(paymentDatePattern, "Invalid Payment Date")) and
+        (JsPath \ "taxablePay").readNullable[Double](paymentAmountValidator) and
+        (JsPath \ "taxYear").readNullable[String](pattern(payeTaxYearPattern, "Invalid Tax Year")) and
+        (JsPath \ "monthlyPeriodNumber").readNullable[String](pattern(monthlyPeriodNumberPattern, "Invalid Monthly Period Number").keepAnd(minLength[String](1)).keepAnd(maxLength[String](2))) and
```

Comment:
Break up long lines (are the minLength and maxLength required here? looks like it's covered by the regex pattern)